include/triton/core/tritonserver.h

```diff
@@ -378,6 +378,31 @@ typedef TRITONSERVER_Error* (*TRITONSERVER_ResponseAllocatorReleaseFn_t)(
 typedef TRITONSERVER_Error* (*TRITONSERVER_ResponseAllocatorStartFn_t)(
     TRITONSERVER_ResponseAllocator* allocator, void* userp);
 
+/// Type for function that is called to query the allocator's preferred memory
+/// type and memory type ID, which will be the attributes of the allocated
+/// buffers if explicitly requested. 'tensor_name' and 'byte_size' are optional
+/// parameters for the allocator. Note that if 'tensor_name' or 'byte_size' is
```

Comment:
Replace "Note that..." with
"As much as possible, the allocator should attempt to return the same memory_type and memory_type_id values that will be returned by the subsequent call to TRITONSERVER_ResponseAllocatorAllocFn_t. But the allocator is not required to do so, and the backend must be capable of handling cases where the values differ."